aioredis/connection.py

```diff
@@ -136,8 +146,9 @@ def __init__(self, reader, writer, *, address, encoding=None,
         self._address = address
         self._loop = loop
         self._waiters = deque()
-        self._parser = parser(protocolError=ProtocolError,
-                              replyError=ReplyError)
+        self._reader.set_parser(
+            parser(protocolError=ProtocolError, replyError=ReplyError)
+        )
```

Comment:
For the sake of compatibility and consistency with the current connection signature, the parser is not set in the `StreamReader` constructor. But if you believe that we can break that let's get rid the `parser` parameter and let's pass as a new argument of the `StreamReader` constructor